### ISSUE TITLE
API extension for alternative file management and rendering (Callbacks)

### ISSUE THREAD

**Issue by Colin-Stone** [zowe/vscode-extension-for-zowe#601]
This issue is to provide the alternative interfaces that would allow rendering and editing of VSAM nodes for example.

This isn't the same as adding a menu but is providing an alternative hook so that the VSAM cluster can be displayed in the explorer view in an alternate way. For example by reducing to one entity. To achieve this type of function we would need to define specific call-back extension points that could re structure a getChildren list 

Added from #692 
Use case is to allow extender application to enhance the contextual information available on a node item to enable them to add further menu options available from their specific extensions.

The logical place to add this context would be during the getChildren method on a node

Add the following methods to the extenders API
addContext(node)
removeContext(node)
addChildrenListener(this)

**Comment by Colin-Stone** [comment]
Not Technical debt the intention of this issue is to identify ways that extenders could perform actions that would compliment or augment what Zowe Explorer does already. 
However the simplest strategy to employ is to have hooks that enable new menus. #651

**Comment by jellypuno** [comment]
@Colin-Stone ok. Can we add a label "API Extender"? Just so we could categorize it. If you like, if not I will leave it unlabeled.

**Comment by Colin-Stone** [comment]
@travatine Where VSAM files are concerned I am wondering if there would be value in combining the three datasets into on item in the explorer. This would require changes to how we get datasets and for now combining would be based on name only. Do you have any thoughts if this would be useful or confusing?.

**Comment by travatine** [comment]
Hi @Colin-Stone ,

I did a quick check of various mainframe products;
 - (3270) ISPF option 3.4 ; the vsam file / data / index are displayed  and can be browsed ( which is not correct - it doesn't make sense to browse an index).
- (3270) IBM File Manager for z/OS (ISPF): the vsam file / data / index are displayed  but only the vsam file can be browsed .
- (Eclipse) IBM Explorer for z/OS : The "Data" and "Index" are hidden ; the "vsam" file appears but can not be opened.

In summary:
Yes, I think it would be very useful to combine the three (vsam file / data / index) into one item.
It'd be great if the contextValue indicated either "vsam" or even "ksds /esds / rrds" if the type is known.

**Comment by Colin-Stone** [comment]
For all I get `dsorg: VS` but the root has `vol: *VSAM*` I can associate the index and data if the root is part of the dsname but if that isn't the case I wouldn't have a way to link them

**Comment by travatine** [comment]
Hi @Colin-Stone , 
That sounds reasonable to me, 
i.e. link if possible otherwise don't.

**Comment by JillieBeanSim** [comment]
This enhancement has had no community activity for 12 months. The issue also has less than 10 up-votes by the community. No action on this enhancement is targeted for the next 2 calendar quarters. Therefore, this enhancement is being closed. If you feel that this enhancement should continue to be available for community up-votes, you may reopen this issue.
